collector/systemd_linux.go

```diff
@@ -57,12 +66,150 @@ func NewSystemdCollector() (Collector, error) {
 		"Whether the system is operational (see 'systemctl is-system-running')",
 		nil, nil,
 	)
+	unitMetrics := metricsMap{
+		"CPUUsageNSec": metric{
+			desc: prometheus.NewDesc(
+				prometheus.BuildFQName(namespace, subsystem, "cpu_usage_nanoseconds_total"),
+				"Total CPU seconds of a unit",
+				[]string{"name"},
+				nil,
+			),
+			valueType: prometheus.CounterValue,
+		},
+		"MemoryCurrent": metric{
+			desc: prometheus.NewDesc(
+				prometheus.BuildFQName(namespace, subsystem, "memory_current"),
+				"Amount of bytes",
+				[]string{"name"},
+				nil,
+			),
+			valueType: prometheus.GaugeValue,
+		},
+		"TasksCurrent": metric{
+			desc: prometheus.NewDesc(
+				prometheus.BuildFQName(namespace, subsystem, "tasks_current"),
+				"amount of tasks. Includes both user processes and kernel threads.",
+				[]string{"name"},
+				nil,
+			),
+			valueType: prometheus.GaugeValue,
+		},
+		"IPIngressBytes": metric{
+			desc: prometheus.NewDesc(
+				prometheus.BuildFQName(namespace, subsystem, "ip_ingress_bytes_total"),
+				"Ingress bytes total",
+				[]string{"name"},
+				nil,
+			),
+			valueType: prometheus.CounterValue,
+		},
+		"IPIngressPackets": metric{
+			desc: prometheus.NewDesc(
+				prometheus.BuildFQName(namespace, subsystem, "ip_ingress_packets_total"),
+				"Ingress packets total",
+				[]string{"name"},
```

Comment:
`name` is too generic, if this is the name of the unit, I would suggest `unit` or `unit_name`.